diff --git signal.c signal.c
index f200634..0b276cc 100644
--- signal.c
+++ signal.c
@@ -806,11 +806,11 @@ check_stack_overflow(int sig, const uintptr_t addr, const ucontext_t *ctx)
 #   endif
 # elif defined __APPLE__
 #   if defined(__LP64__)
-    const uintptr_t sp = mctx->__ss.__rsp;
-    const uintptr_t bp = mctx->__ss.__rbp;
+    const uintptr_t sp = mctx->ss.rsp;
+    const uintptr_t bp = mctx->ss.rbp;
 #   else
-    const uintptr_t sp = mctx->__ss.__esp;
-    const uintptr_t bp = mctx->__ss.__ebp;
+    const uintptr_t sp = mctx->ss.esp;
+    const uintptr_t bp = mctx->ss.ebp;
 #   endif
 # elif defined __FreeBSD__
 #   if defined(__amd64__)
diff --git vm_dump.c vm_dump.c
index 4ed7688..75b133b 100644
--- vm_dump.c
+++ vm_dump.c
@@ -13,6 +13,7 @@
 #include "addr2line.h"
 #include "vm_core.h"
 #include "iseq.h"
+#include <ucontext.h>
 
 /* see vm_insnhelper.h for the values */
 #ifndef VMDEBUG
@@ -842,7 +843,7 @@ print_machine_register(size_t reg, const char *reg_name, int col_count, int max_
 # ifdef __linux__
 #   define dump_machine_register(reg) (col_count = print_machine_register(mctx->gregs[REG_##reg], #reg, col_count, 80))
 # elif defined __APPLE__
-#   define dump_machine_register(reg) (col_count = print_machine_register(mctx->__ss.__##reg, #reg, col_count, 80))
+#   define dump_machine_register(reg) (col_count = print_machine_register(mctx->ss.reg, #reg, col_count, 80))
 # endif
 
 static void
